core/src/main/java/org/apache/struts2/dispatcher/multipart/AbstractMultiPartRequest.java

```diff
@@ -171,4 +171,8 @@ protected String getCanonicalName(final String originalFileName) {
         return fileName;
     }
 
+    protected String sanitizeNewlines(String before) {
+        return before.replaceAll("[\n\r]", "_");
```

Comment:
```suggestion
        return before.replaceAll("\R", "_");
```

maybe use this as more concise way to represent line breaks?

>Any Unicode linebreak sequence, is equivalent to \u000D\u000A|[\u000A\u000B\u000C\u000D\u0085\u2028\u2029]